### PR TITLE
Migrate pipelines to 1ES

### PR DESCRIPTION
**Pull request by habbes**
This PR migrates the build pipelines to 1ES.

### CODE REVIEW

**Review by xuzhg — APPROVED** [review]

File: pipelines/azure_pipelines_nightly.yml [hunk 1]
```diff
@@ -1,109 +1,288 @@
...
+            platform: $(BuildPlatform)
+            configuration: $(BuildConfiguration)
+        - task: VSBuild@1
+          displayName: Build solution sln\WebApiOData.E2E.AspNet.sln
+          inputs:
+            solution: sln\WebApiOData.E2E.AspNet.sln
+            platform: $(BuildPlatform)
+            configuration: $(BuildConfiguration)
+        - task: DotNetCoreCLI@2
+          displayName: 'build Microsoft.AspNetCore.OData.csproj '
```
Comment (gathogojr) [inline pipelines/azure_pipelines_nightly.yml:88]: nit:
```suggestion
          displayName: 'Build Microsoft.AspNetCore.OData.csproj '
```
Most other `displayName` start with upper case

File: pipelines/azure_pipelines_nightly.yml [hunk 2]
```diff
@@ -1,109 +1,288 @@
...
+          inputs:
+            projects: $(Build.SourcesDirectory)\src\Microsoft.AspNetCore.OData\Microsoft.AspNetCore.OData.csproj
+            arguments: --configuration $(BuildConfiguration) --no-incremental
+        - task: DotNetCoreCLI@2
+          displayName: build .NET Core Unit test project
```
Comment (gathogojr) [inline pipelines/azure_pipelines_nightly.yml:93]: See my comment here https://github.com/OData/WebApi/pull/2841/files#r1619893321

File: pipelines/azure_pipelines_nightly.yml [hunk 3]
```diff
@@ -1,109 +1,288 @@
...
+          inputs:
+            projects: $(Build.SourcesDirectory)\test\UnitTest\Microsoft.AspNetCore.OData.Test\Microsoft.AspNetCore.OData.Test.csproj
+            arguments: --configuration $(BuildConfiguration) --no-incremental
+        - task: DotNetCoreCLI@2
+          displayName: build .NET Core E2E test project
```
Comment (gathogojr) [inline pipelines/azure_pipelines_nightly.yml:98]: See my comment here https://github.com/OData/WebApi/pull/2841/files#r1619893321

File: pipelines/azure_pipelines_nightly.yml [hunk 4]
```diff
@@ -1,109 +1,288 @@
...
+          inputs:
+            projects: $(Build.SourcesDirectory)\test\E2ETest\Microsoft.Test.E2E.AspNet.OData\Build.AspNetCore\Microsoft.Test.E2E.AspNetCore.OData.csproj
+            arguments: --configuration $(BuildConfiguration) --no-incremental
+        - task: DotNetCoreCLI@2
+          displayName: build .NET Core 3x E2E test project
```
Comment (gathogojr) [inline pipelines/azure_pipelines_nightly.yml:103]: See my comment here https://github.com/OData/WebApi/pull/2841/files#r1619893321

File: pipelines/azure_pipelines_nightly.yml [hunk 5]
```diff
@@ -1,109 +1,288 @@
...
+          inputs:
+            command: test
+            projects: $(Build.SourcesDirectory)\test\E2ETest\Microsoft.Test.E2E.AspNet.OData\Build.AspNetCore\Microsoft.Test.E2E.AspNetCore.OData.csproj
+            arguments: --configuration $(BuildConfiguration) --no-build
+        - task: DotNetCoreCLI@2
+          displayName: Core E2E 3x Tests (Microsoft.Test.E2E.AspNetCore3x.OData.csproj)
+          inputs:
+            command: test
+            projects: $(Build.SourcesDirectory)\test\E2ETest\Microsoft.Test.E2E.AspNet.OData\Build.AspNetCore3x\Microsoft.Test.E2E.AspNetCore3x.OData.csproj
+            arguments: --configuration $(BuildConfiguration) --no-build  --filter FullyQualifiedName!=Microsoft.Test.E2E.AspNet.OData.QueryComposition.SelectExpandEFTests.QueryForLongSelectList -v diag
```
Comment (gathogojr) [inline pipelines/azure_pipelines_nightly.yml:161]: Would you know why we filter out this particular test(s)?

File: pipelines/azure_pipelines.yml [hunk 6]
```diff
@@ -3,27 +3,139 @@ trigger:
...
+            platform: $(BuildPlatform)
+            configuration: $(BuildConfiguration)
+        - task: VSBuild@1
+          displayName: Build solution sln\WebApiOData.E2E.AspNet.sln
+          inputs:
+            solution: sln\WebApiOData.E2E.AspNet.sln
+            platform: $(BuildPlatform)
+            configuration: $(BuildConfiguration)
+        - task: DotNetCoreCLI@2
+          displayName: 'build Microsoft.AspNetCore.OData.csproj '
```
Comment (gathogojr) [inline pipelines/azure_pipelines.yml:73]: See my comment here https://github.com/OData/WebApi/pull/2841/files#r1619893321

File: pipelines/azure_pipelines.yml [hunk 7]
```diff
@@ -3,27 +3,139 @@ trigger:
...
+          inputs:
+            projects: $(Build.SourcesDirectory)\src\Microsoft.AspNetCore.OData\Microsoft.AspNetCore.OData.csproj
+            arguments: --configuration $(BuildConfiguration) --no-incremental
+        - task: DotNetCoreCLI@2
+          displayName: build .NET Core Unit test project
```
Comment (gathogojr) [inline pipelines/azure_pipelines.yml:78]: See my comment here https://github.com/OData/WebApi/pull/2841/files#r1619893321

File: pipelines/azure_pipelines.yml [hunk 8]
```diff
@@ -3,27 +3,139 @@ trigger:
...
+          inputs:
+            projects: $(Build.SourcesDirectory)\test\UnitTest\Microsoft.AspNetCore.OData.Test\Microsoft.AspNetCore.OData.Test.csproj
+            arguments: --configuration $(BuildConfiguration) --no-incremental
+        - task: DotNetCoreCLI@2
+          displayName: build .NET Core E2E test project
```
Comment (gathogojr) [inline pipelines/azure_pipelines.yml:83]: See my comment here https://github.com/OData/WebApi/pull/2841/files#r1619893321

File: pipelines/azure_pipelines.yml [hunk 9]
```diff
@@ -3,27 +3,139 @@ trigger:
...
+          inputs:
+            projects: $(Build.SourcesDirectory)\test\E2ETest\Microsoft.Test.E2E.AspNet.OData\Build.AspNetCore\Microsoft.Test.E2E.AspNetCore.OData.csproj
+            arguments: --configuration $(BuildConfiguration) --no-incremental
+        - task: DotNetCoreCLI@2
+          displayName: build .NET Core 3x E2E test project
```
Comment (gathogojr) [inline pipelines/azure_pipelines.yml:88]: See my comment here https://github.com/OData/WebApi/pull/2841/files#r1619893321

File: pipelines/azure_pipelines.yml [hunk 10]
```diff
@@ -3,27 +3,139 @@ trigger:
...
+          inputs:
+            command: test
+            projects: $(Build.SourcesDirectory)\test\E2ETest\Microsoft.Test.E2E.AspNet.OData\Build.AspNetCore\Microsoft.Test.E2E.AspNetCore.OData.csproj
+            arguments: --configuration $(BuildConfiguration) --no-build
+        - task: DotNetCoreCLI@2
+          displayName: Core E2E 3x Tests (Microsoft.Test.E2E.AspNetCore3x.OData.csproj)
+          inputs:
+            command: test
+            projects: $(Build.SourcesDirectory)\test\E2ETest\Microsoft.Test.E2E.AspNet.OData\Build.AspNetCore3x\Microsoft.Test.E2E.AspNetCore3x.OData.csproj
+            arguments: --configuration $(BuildConfiguration) --no-build --filter FullyQualifiedName!=Microsoft.Test.E2E.AspNet.OData.QueryComposition.SelectExpandEFTests.QueryForLongSelectList -v diag
```
Comment (gathogojr) [inline pipelines/azure_pipelines.yml:141]: Out of scope, but would you know why we filter out this particular test(s)?

File: pipelines/azure_pipelines.yml [hunk 11]
```diff
@@ -3,27 +3,139 @@ trigger:
...
+          displayName: 'Core Unit Tests (Microsoft.AspNetCore.OData.Test.csproj) '
+          inputs:
+            command: test
+            projects: $(Build.SourcesDirectory)\test\UnitTest\Microsoft.AspNetCore.OData.Test\Microsoft.AspNetCore.OData.Test.csproj
+            arguments: --configuration $(BuildConfiguration) --no-build
+        - task: VSTest@2
+          displayName: Classic E2E Tests (Microsoft.Test.E2E.AspNet.OData.dll)
+          inputs:
+            testAssemblyVer2: '**\bin\**\E2ETest\**\Microsoft.Test.E2E.AspNet.OData.dll'
+            testFiltercriteria: (DisplayName!=Microsoft.Test.E2E.AspNet.OData.ODataPathHandler.UnicodeRouteTests_Todoü.CRUDEntitySetShouldWork)
```
Comment (gathogojr) [inline pipelines/azure_pipelines.yml:129]: Out of scope, but would you know why we filter out this particular test(s)?

**Review by gathogojr — APPROVED** [review]
Left some comments; otherwise LGTM

File: pipelines/azure_pipelines.yml (hunk 11 above)
Comment (habbes) [reply pipelines/azure_pipelines.yml:129]: I'm not sure. My initial guess woulld be to reduce runtime.

**Pull request merged**